src/vm/builtins/builtin_runner/builtin_runner.zig

```diff
@@ -126,7 +126,7 @@ pub const BuiltinRunner = union(enum) {
             .RangeCheck => |range_check| range_check.deduceMemoryCell(address, memory),
             .Keccak => |*keccak| try keccak.deduceMemoryCell(allocator, address, memory),
             .Signature => |signature| signature.deduceMemoryCell(address, memory),
-            .Poseidon => |poseidon| poseidon.deduceMemoryCell(address, memory),
+            .Poseidon => |*poseidon| try poseidon.deduceMemoryCell(allocator, address, memory),
```

Comment:
one more ask: see in this change where we added a `getMemorySegmentAddresses` for the builtin_runner

we want to add this completed poseidons impl of `getMemorySegmentAddresses` there, then lgtm!